src/xmipp/libraries/data/monogenic.h

```diff
@@ -0,0 +1,172 @@
+
+	// STATISTICSINOUTBINARYMASK2: Estimates the staticstics of a single
+	// map. The signal statistics are obtained by mean of a binary mask.
+	// The results are the mean and variance of noise and signal, as well
+	// as the number of voxels of signal and noise NS and NN respectively.
+	// The thr95 represents the percentile 95 of noise distribution.
+	void statisticsInOutBinaryMask2(const MultidimArray<double> &volS,
+			MultidimArray<int> &mask, double &meanS, double &sdS2,
+			double &meanN, double &sdN2, double &significance, double &thr95, double &NS, double &NN);
+
+	// SETLOCALRESOLUTIONHALFMAPS: Set the local resolution of a voxel, by
+	// determining if the monogenic amplitude "amplitudeMS" is higher than
+	// the threshold of noise "thresholdNoise". Thus the local resolution
+	// map "plocalResolution" is set, with the resolution value "resolution"
+	// or with "resolution_2". "resolution" is the resolution for with
+	// the hypohtesis test is carried out, and "resolution_2" is the resolution
+	// of the analisys two loops ago (see monores method)
+	void setLocalResolutionHalfMaps(const MultidimArray<double> &amplitudeMS,
+			MultidimArray<int> &pMask, MultidimArray<double> &plocalResolutionMap,
+			double &thresholdNoise, double &resolution, double &resolution_2);
+
+	// SETLOCALRESOLUTIONMAP: Set the local resolution of a voxel, by
+	// determining if the monogenic amplitude "amplitudeMS" is higher than
+	// the threshold of noise "thresholdNoise". Thus the local resolution
+	// map "plocalResolution" is set, with the resolution value "resolution"
+	// or with "resolution_2". "resolution" is the resolution for with
+	// the hypohtesis test is carried out, and "resolution_2" is the resolution
+	// of the analisys two loops ago (see monores method)
+	void setLocalResolutionMap(const MultidimArray<double> &amplitudeMS,
+		MultidimArray<int> &pMask, MultidimArray<double> &plocalResolutionMap,
+		double &thresholdNoise, double &resolution, double &resolution_2);
+
+	// MONOGENICAMPLITUDE_3D_FOURIER: Given the fourier transform of a map
+	// "myfftV", this function computes the monogenic amplitude "amplitude" 
+	// iu is the inverse of the frequency in Fourier space.
+	void monogenicAmplitude_3D_Fourier(const MultidimArray< std::complex<double> > &myfftV,
+			MultidimArray<double> iu, MultidimArray<double> &amplitude, int numberOfThreads);
+
+	//ADDNOISE: This function add gaussian with mean = double mean and standard deviation
+	// equal to  double stddev to a map given by "vol"
+	void addNoise(MultidimArray<double> &vol, double mean, double stddev);
+
+	//CREATEDATATEST: This function generates fringe pattern (rings) map returned 
+	// as a multidimArray, with  dimensions given by xdim, ydim and zdim. The 
+	// wavelength of the pattern is given by double wavelength
+	MultidimArray<double> createDataTest(size_t xdim, size_t ydim, size_t zdim,
+		double wavelength, double mean, double sigma);
+
```

Comment:
Why don't you have class fields, e.g. `fftVRiesz`?

Reply:
I will think about that, for next pull request. It think it can be a good idea